### PR TITLE
adding hasIdentified to do better checks before tracking an event

### PR DESCRIPTION
**Pull request by usmanity**
This new way of checking for an identified helps Segment customers send anonymous user IDs when users are not identified with an actual user id yet.

### CODE REVIEW

**Comment by usmanity** [comment]
@ndhoule hi, circle Ci is complaining I don't have stubs for 2 of my tests but they are present. Can you please take a look why they're not passing? Does it give this error even when something else could be wrong? Thanks.

**Comment by usmanity** [comment]
@ndhoule 
@sperand-io 

Hi, I've been testing this locally and the tests pass but when I circle runs the tests, they are failing, any idea what can be different about how I run my tests? 

I am using the command 
`make test-browser`

Screenshot of my tests in Chrome
screenshot 2015-08-05 13 52 49

**Comment by ndhoule** [comment]
Hey @usmanity—are your tests passing while running `make test`?

**Comment by usmanity** [comment]
@ndhoule just ran that and nope they are not passing. I get the same result as Circle. Any idea how I can debug this to see what could be causing the tests to break? In the browser, it's pretty easy to find the issue using the dev tools.

**Comment by usmanity** [comment]
@ndhoule just ran that and nope they are not passing. I get the same result as Circle. Any idea how I can debug this to see what could be causing the tests to break? In the browser, it's pretty easy to find the issue using the dev tools.

**Comment by usmanity** [comment]
Please merge when possible. Thank you!

File: lib/index.js [hunk 1]
```diff
@@ -103,6 +94,9 @@ Outbound.prototype.identify = function(identify) {
  */
 
 Outbound.prototype.track = function(track) {
+  if (!this.hasIdentified()) {
+    this.identify(track.userId() || track.anonymousId());
```
Comment (sperand-io) [inline lib/index.js]: identify takes a wrapped `identify` facade object, not a string. this will fail

Comment (usmanity) [inline lib/index.js]: @sperand-io will the `identify` facade object be available to me in a `track` call?

Comment (usmanity) [inline lib/index.js]: Or should I just pass in an object like:

```
  this.identify({
    userId: track.userId()
  })
```

**Comment by sperand-io** [comment]
Hey @usmanity thanks for the pull! Before accepting, we'd really appreciate a more thorough explanation of the business case for the change here. Does outbound have the notion of an anonymous user? 

Or will we just be sending track calls attributed to anonymous users who can then be aliased? Generally we try not to conflate `identify` with anonymous users, opting instead to simply send track calls anonymously. Let me know if y'all want to hop on a quick call or something to get this sorted.

**Comment by usmanity** [comment]
@sperand-io hi, this change is required because our library will reject track calls unless a user has been identify as anonymous user or an actual user with an ID. If it is an anonymous user, the user can be aliased.

File: lib/index.js (hunk 1 above)
Comment (sperand-io) [inline lib/index.js]: > @sperand-io will the identify facade object be available to me in a track call?

Unfortunately no, but you can construct one manually like so:

```
// add segmentio/analytics.js-integration dep to component.json
var Identify = require('facade').Identify;

// ...

var user = new Identify({ userId: track.userId() || track.anonymousId() });
this.identify(user);
```

**Comment by sperand-io** [comment]
Gotcha, thanks for the explanation! I commented inline re: triggering the identify with a facade-wrapped obj, lmk if that makes sense!

**Comment by usmanity** [comment]
@sperand-io I've added the code sample in `index.js` along with the `segmentio/analytics.js-integration` in dependencies file. but I am not able to find the `facade` module, is that another module I should include in `component.json`?

**Comment by sperand-io** [comment]
@usmanity yea, that should do it! --

```
  "dependencies": {
     // ...
    "segmentio/facade": "^1.4.8",
  },
```

**Comment by usmanity** [comment]
@sperand-io thanks! updated and tests are passing

File: lib/index.js [hunk 2]
```diff
@@ -51,16 +53,6 @@ Outbound.prototype.initialize = function() {
...
-/**
- * Loaded?
- *
- * @api private
- * @return {boolean}
- */
-
-Outbound.prototype.loaded = function() {
-  return window.outbound;
-};
```
Comment (sperand-io) [inline lib/index.js:63]: why did we delete this?

File: test/index.test.js [hunk 3]
```diff
@@ -64,12 +66,6 @@ describe('Outbound', function() {
     });
   });
 
-  describe('loading', function() {
-    it('should load', function(done) {
-      analytics.load(outbound, done);
-    });
-  });
```
Comment (sperand-io) [inline test/index.test.js:71]: ?

File: lib/index.js (hunk 2 above)
Comment (usmanity) [inline lib/index.js:63]: not needed anymore, I had this just to test for the library being loaded in the tests, never actually used it in the outbound integration.

File: test/index.test.js (hunk 3 above)
Comment (usmanity) [inline test/index.test.js:71]: is this a required method for integrations to include when integrating with segment?

Comment (sperand-io) [inline test/index.test.js:71]: we like to leave it in to ensure in testing that the integration does load as expected. it's not required for the integration to work but is definitely a standard in all our a.js integrations, so I just wondered if you removed it for any particular reason? like if that test was failing, etc

Comment (usmanity) [inline test/index.test.js:71]: it was breaking the tests but after attempting to fix them, I realized that we were testing it for the sake of it being part of our code and not that we needed to use it. @sperand-io

**Comment by usmanity** [comment]
Any updates on this pull request @sperand-io?

**Pull request closed without merging**